website/orders/api/v1/views.py

```diff
@@ -269,15 +278,19 @@ def patch(self, request, **kwargs):
         """
         shift = kwargs.get("shift")
         time_minutes = request.data.get("minutes", 5)
-        increase_shift_time(shift, time_minutes)
-        return Response(status=status.HTTP_200_OK, data=ShiftSerializer(shift, context={"request": request}).data)
+        try:
+            increase_shift_time(shift, time_minutes)
+            return Response(status=status.HTTP_200_OK, data=ShiftSerializer(shift, context={"request": request}).data)
+        except ValueError as e:
+            return Response(status=status.HTTP_403_FORBIDDEN, data={"detail": e.__str__()})
```

Comment:
Why is this changed?

Reply:
Increasing the shift time can actually now throw an exception (due to shifts being finalized). Therefor we need to check it in the view and return a 403 if the shift can't be changed. I now see that this is the wrong exception caught though (it should be a ValidationError).

Reply:
This is now a ValidationError

Reply:
Idem, I think this can raise an exception